### PR TITLE
EVG-6369 add GET /roles

### PR DESCRIPTION
**Pull request by john-m-liu**

### CODE REVIEW

File: rest/route/user_test.go [hunk 1]
```diff
@@ -8,6 +8,8 @@ import (
 	"testing"
 	"time"
 
+	"github.com/stretchr/testify/assert"
+
```
Comment (ablack12) [inline rest/route/user_test.go]: nit: whitespace

File: rest/route/service.go [hunk 2]
```diff
@@ -110,6 +110,7 @@ func AttachHandler(app *gimlet.APIApp, opts HandlerOpts) {
 	app.AddRoute("/projects/{project_id}/test_stats").Version(2).Get().Wrap(checkUser).RouteHandler(makeGetProjectTestStats(sc))
 	app.AddRoute("/projects/{project_id}/task_stats").Version(2).Get().Wrap(checkUser).RouteHandler(makeGetProjectTaskStats(sc))
 	app.AddRoute("/projects/{project_id}/task_reliability").Version(2).Get().Wrap(checkUser).RouteHandler(makeGetProjectTaskReliability(sc))
+	app.AddRoute("/roles").Version(2).Get().Wrap(checkUser).RouteHandler(makeGetAllRolesHandler(sc))
```
Comment (ablack12) [inline rest/route/service.go]: Is the idea that any user should be able to see all the roles? Is there any reason these might be private?

Comment (john-m-liu) [reply rest/route/service.go]: Some roles might be scoped to private projects, so we don't want to reveal the fact that they exist

File: rest/route/user_test.go (hunk 1 above)
Comment (john-m-liu) [reply rest/route/user_test.go]: fixed

**Pull request merged**